datadog_lambda/tracing.py

```diff
@@ -377,13 +480,394 @@ def set_dd_trace_py_root(trace_context_source, merge_xray_traces):
+    span = tracer.trace("aws.kinesis", **args)
+    if span:
+        span.set_tags(tags)
+    span.start = request_time_epoch
+    return span
+
+
+def create_inferred_span_from_dynamodb_event(event, context):
+    event_record = get_first_record(event)
+    event_source_arn = event_record["eventSourceARN"]
+    table_name = event_source_arn.split("/")[1]
+    dynamodb_message = event_record["dynamodb"]
+    tags = {
+        "operation_name": "aws.dynamodb",
+        "resource_names": table_name,
+        "tablename": table_name,
+        "event_source_arn": event_source_arn,
+        "event_id": event_record["eventID"],
+        "event_name": event_record["eventName"],
+        "event_version": event_record["eventVersion"],
+        "stream_view_type": dynamodb_message["StreamViewType"],
+        "size_bytes": str(dynamodb_message["SizeBytes"]),
+    }
+    InferredSpanInfo.set_tags(tags, synchronicity="async", tag_source="self")
+    request_time_epoch = event_record["dynamodb"]["ApproximateCreationDateTime"]
+    args = {
+        "service": "dynamodb",
+        "resource": table_name,
+        "span_type": "web",
+    }
+    tracer.set_tags({"_dd.origin": "lambda"})
+    span = tracer.trace("aws.dynamodb", **args)
+    if span:
+        span.set_tags(tags)
+
+    span.start = int(request_time_epoch)
+    return span
+
+
+def create_inferred_span_from_s3_event(event, context):
+    event_record = get_first_record(event)
+    bucket_name = event_record["s3"]["bucket"]["name"]
+    tags = {
+        "operation_name": "aws.s3",
+        "resource_names": bucket_name,
+        "event_name": event_record["eventName"],
+        "bucketname": bucket_name,
+        "bucket_arn": event_record["s3"]["bucket"]["arn"],
+        "object_key": event_record["s3"]["object"]["key"],
+        "object_size": str(event_record["s3"]["object"]["size"]),
+        "object_etag": event_record["s3"]["object"]["eTag"],
+    }
+    InferredSpanInfo.set_tags(tags, synchronicity="async", tag_source="self")
+    dt_format = "%Y-%m-%dT%H:%M:%S.%fZ"
+    timestamp = event_record["eventTime"]
+    dt = datetime.strptime(timestamp, dt_format)
+
+    args = {
+        "service": "s3",
+        "resource": bucket_name,
+        "span_type": "web",
+    }
+    tracer.set_tags({"_dd.origin": "lambda"})
+    span = tracer.trace("aws.s3", **args)
+    if span:
+        span.set_tags(tags)
+    span.start = dt.replace(tzinfo=timezone.utc).timestamp()
+    return span
+
+
+def create_inferred_span_from_eventbridge_event(event, context):
+    source = event["source"]
+    tags = {
+        "operation_name": "aws.eventbridge",
+        "resource_names": source,
+        "detail_type": event["detail-type"],
+    }
+    InferredSpanInfo.set_tags(
+        tags,
+        synchronicity="async",
+        tag_source="self",
+    )
+    dt_format = "%Y-%m-%dT%H:%M:%SZ"
+    timestamp = event["time"]
+    dt = datetime.strptime(timestamp, dt_format)
```

Comment:
:/ I wish we had millisecond resolution here, but this will have to do for now

Reply:
Yeah, we need a figure out a solution for this and potentially other spans down the road with a similar issue.